acceptance-tests/src/test/java/tech/pegasys/web3signer/tests/WatermarkRepairSubCommandAcceptanceTest.java

```diff
@@ -106,6 +105,131 @@ void allLowWatermarksAreUpdated(@TempDir final Path testDirectory) throws URISyn
     assertThat(validator2.get("target_epoch")).isEqualTo(epoch);
   }
 
+  @Test
+  void highWatermarkPreventsImportsAndSignatures(@TempDir final Path testDirectory)
+      throws URISyntaxException, JsonProcessingException {
+
+    /*
+       1. Set high watermark
+       2. Importing slashing data beyond high watermark prevents import
+       3. Signing beyond high watermark is prevented
+       4. Resetting high watermark to lower value fails due to low watermark conflict
+       5. Removing high watermark allows slashing import
+    */
+
+    setupSigner(testDirectory);
+
+    // Import validator1's data to set the GVR
+    importSlashingProtectionData(testDirectory, "slashing/slashingImport.json");
+
+    final SignerConfigurationBuilder commandConfig = commandConfig();
+
+    // Set high watermark between the two slashing import entries to prevent second entry from being
+    // imported
+    long highWatermarkSlot = 19998L;
+    long highWatermarkEpoch = 7L;
+    executeSubcommand(
+        commandConfig, new WatermarkRepairParameters(highWatermarkSlot, highWatermarkEpoch, true));
+
+    assertGetHighWatermarkEquals(highWatermarkSlot, highWatermarkEpoch);
+
+    // Import with second entry beyond the high watermark
+    importSlashingProtectionData(testDirectory, "slashing/slashingImport_two_entries.json");
+
+    Map<Object, List<Map<String, Object>>> watermarks = getWatermarks();
+    assertThat(watermarks).hasSize(1);
+    Map<String, Object> validator1 =
+        watermarks
+            .get(
+                "0x8f3f44b74d316c3293cced0c48c72e021ef8d145d136f2908931090e7181c3b777498128a348d07b0b9cd3921b5ca537")
+            .get(0);
+    assertThat(validator1.get("slot")).isEqualTo(BigDecimal.valueOf(12345));
+    assertThat(validator1.get("source_epoch")).isEqualTo(BigDecimal.valueOf(5));
+    assertThat(validator1.get("target_epoch")).isEqualTo(BigDecimal.valueOf(6));
+
+    // validator2 is not imported due to high watermark
+    assertThat(
```

Comment:
This is only checking that the watermarks didn't get updated. We should check that signed attestations and blocks data are not updated

Reply:
done